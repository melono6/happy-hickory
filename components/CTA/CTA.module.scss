@import "../../styles/variables.scss";

.CTA {
  font-size: 1.6rem;
  letter-spacing: -0.025rem;
  line-height: 2.4rem;
  text-decoration: none;
  color: $play-black;

  &:hover,
  &:focus {
    .ButtonStyleOne {
      background-color: $play-green;
    }

    .CTAText {
      &::after {
        visibility: visible;
        width: 100%;
      }
    }
  }
}

.CTAText {
  @include hyperlink-initial-state($play-black);
}

.Primary {
  display: flex;
  align-items: center;
}

.ButtonStyleOne {
  width: 5.6rem;
  min-width: 5.6rem;
  height: 5.6rem;
  background-color: $play-black;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  margin-right: 1.6rem;
  border-radius: 9999rem;
  transition: background-color 0.3s ease-in-out;
}

.DotsWrapper {
  position: relative;
  height: 2.4rem;
  width: 1rem;
  display: flex;
  justify-content: space-between;
  flex-direction: column;
}

.Dot {
  background-color: $white;
  border-radius: 9999rem;
  width: 0.4rem;
  height: 0.4rem;

  &:nth-child(1),
  &:nth-child(5) {
    margin-left: 0;
  }
  &:nth-child(2),
  &:nth-child(4) {
    margin-left: 0.4rem;
  }
  &:nth-child(3) {
    margin-left: 0.8rem;
  }
}

.ExternalLinkIcon {
  width: 1.6rem;
  height: 1.6rem;
  margin-left: 2rem;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  margin-left: 1.6rem;
  margin-bottom: 0.4rem;

  svg {
    width: 100%;
    height: 100%;
  }
}
