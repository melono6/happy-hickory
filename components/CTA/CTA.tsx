import React from "react";
import Link from "next/link";
import styles from "./CTA.module.scss";
import { styleNames } from "@zepp/utils";
import ExternalLinkIcon from "../../public/image/svg/external-link-icon.svg";

export type CTAProps = {
  text: string;
  href: string;
  theme?: "Primary" | "Secondary" | string[];
  type?: "CTA" | "submit" | "reset";
  target?: "_blank" | "_self";
};

const PrimaryDots = () => (
  <span className={styles.ButtonStyleOne}>
    <span className={styles.DotsWrapper}>
      <span className={styles.Dot} />
      <span className={styles.Dot} />
      <span className={styles.Dot} />
      <span className={styles.Dot} />
      <span className={styles.Dot} />
    </span>
  </span>
);

export const CTA = ({ text, href, theme, type = "CTA", target }: CTAProps) => {
  const isPrimary = theme === "Primary";
  const isExernal = target === "_blank";
  return (
    <Link href={href} passHref>
      <a
        className={styleNames(styles, ["CTA", theme])}
        type={type}
        target={target}
      >
        {isPrimary && <PrimaryDots />}
        <span className={styles.CTAText}>{text}</span>
        {isExernal && (
          <span className={styles.ExternalLinkIcon}>
            <ExternalLinkIcon />
          </span>
        )}
      </a>
    </Link>
  );
};

export default CTA;
