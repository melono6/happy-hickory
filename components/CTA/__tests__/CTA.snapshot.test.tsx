import React from "react";
import CTA from "../CTA";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<CTA href="/" text="Some Text" />).toJSON();
  expect(tree).toMatchSnapshot();
});
