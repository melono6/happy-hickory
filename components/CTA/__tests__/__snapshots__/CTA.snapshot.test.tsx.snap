// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  className=""
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
  type="CTA"
>
  <span>
    Some Text
  </span>
</a>
`;
